facebook-commerce.php

```diff
@@ -4238,51 +4238,67 @@ function fb_duplicate_product_reset_meta( $to_delete ) {
 		return $to_delete;
 	}
 
+
 	/**
 	 * Helper function to update FB visibility.
+	 *
+	 * @param int $product_id product ID
+	 * @param string $visibility visibility
 	 */
-	function update_fb_visibility( $wp_id, $visibility ) {
+	function update_fb_visibility( $product_id, $visibility ) {
 
 		// bail if the plugin is not configured properly
 		if ( ! $this->is_configured() || ! $this->get_product_catalog_id() ) {
 			return;
 		}
 
-		$woo_product = new WC_Facebook_Product( $wp_id );
+		$product = wc_get_product( $product_id );
 
-		if ( ! $woo_product->exists() ) {
-			// This function can be called for non-woo products.
+		// bail if product isn't found
+		if ( ! $product instanceof \WC_Product ) {
 			return;
 		}
 
-		$products = WC_Facebookcommerce_Utils::get_product_array( $woo_product );
-		foreach ( $products as $item_id ) {
-			$fb_product_item_id = $this->get_product_fbid(
-				self::FB_PRODUCT_ITEM_ID,
-				$item_id
-			);
+		$should_set_visible = $visibility === self::FB_SHOP_PRODUCT_VISIBLE;
+
+		if ( ! $product->is_type( 'variable' ) ) {
+
+			$fb_product_item_id = $this->get_product_fbid( self::FB_PRODUCT_ITEM_ID, $product_id );
 
 			if ( ! $fb_product_item_id ) {
-				WC_Facebookcommerce_Utils::fblog(
-					$fb_product_item_id . " doesn't exist but underwent a visibility transform.",
-					array(),
-					true
-				);
-				  continue;
+				\WC_Facebookcommerce_Utils::fblog( $fb_product_item_id . " doesn't exist but underwent a visibility transform.", [], true );
+				 return;
 			}
-			$result = $this->check_api_result(
-				$this->fbgraph->update_product_item(
-					$fb_product_item_id,
-					array( 'visibility' => $visibility )
-				)
-			);
-			if ( $result ) {
 
-				$is_visible = $visibility === self::FB_SHOP_PRODUCT_VISIBLE;
+			$set_visibility = $this->fbgraph->update_product_item( $fb_product_item_id, [ 'visibility' => $visibility ] );
+
+			if ( $this->check_api_result( $set_visibility ) ) {
+				Products::set_product_visibility( $product, $should_set_visible );
+			}
+
+		} else {
+
+		    // parent product
+			Products::set_product_visibility( $product, $should_set_visible );
 
-				update_post_meta( $item_id, Products::VISIBILITY_META_KEY, wc_bool_to_string( $is_visible ) );
-				update_post_meta( $wp_id, Products::VISIBILITY_META_KEY, wc_bool_to_string( $is_visible ) );
+			$product_ids = [ $product_id ];
```

Comment:
should we still set visibility for variable parent product and pass that to the array of IDs to sync?

I believe this is what was happening before this PR, in this method's context

Reply:
@wvega, I'd like your input on this one. When updating visibility for a variable product, we schedule the update for the variations. What about the parent product? Should we also schedule it or run it immediately as we do for simple products?

Reply:
@unfulvio @deivamagalhaes based on what I've seen while working with the API, product groups, which are used to represent the parent product for variable product, don't have visibility property on Facebook. I think we should **not** add the parent product's ID to the list of IDs to update or create.

The previous version of the method was using [`WC_Facebookcommerce_Utils::get_product_array( $woo_product )`](https://github.com/facebookincubator/facebook-for-woocommerce/blob/2d0679bb3a6246698d1813a816e6e74c3e1d30d6/includes/fbutils.php#L159-L177) to get the IDs, which doesn't include the ID of the parent in the case of a variable product.

Reply:
Perfect, thanks, @wvega, for clarifying! 🙏